File: shared/src/types/child.ts
export interface Child {
  id: string;
  parent: string;
  first_name: string;
  last_name: string;
  date_of_birth: string;
  allowance_amount: number;
  allowance_frequency: 'weekly' | 'monthly';
  points: number;
  is_frozen: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateChildRequest {
  first_name: string;
  last_name: string;
  date_of_birth: string;
  allowance_amount: number;
  allowance_frequency: 'weekly' | 'monthly';
}

export interface UpdateChildRequest {
  first_name?: string;
  last_name?: string;
  allowance_amount?: number;
  allowance_frequency?: 'weekly' | 'monthly';
}

export interface SpendingLimits {
  daily_limit: number;
  weekly_limit: number;
  monthly_limit: number;
  per_transaction_limit: number;
}

export interface UpdateSpendingLimitsRequest {
  daily_limit?: number;
  weekly_limit?: number;
  monthly_limit?: number;
  per_transaction_limit?: number;
}

export interface TransferRequest {
  child_id: string;
  amount: number;
  description?: string;
}

export interface MoneyRequest {
  id: string;
  child: string;
  amount: number;
  reason: string;
  status: 'pending' | 'approved' | 'declined';
  created_at: string;
}

export interface CreateMoneyRequestRequest {
  amount: number;
  reason: string;
}

File: shared/dart/lib/child.dart
class Child {
  final String id;
  final String parent;
  final String firstName;
  final String lastName;
  final DateTime dateOfBirth;
  final double allowanceAmount;
  final String allowanceFrequency;
  final int points;
  final DateTime createdAt;
  final DateTime updatedAt;

  Child({
    required this.id,
    required this.parent,
    required this.firstName,
    required this.lastName,
    required this.dateOfBirth,
    required this.allowanceAmount,
    required this.allowanceFrequency,
    required this.points,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Child.fromJson(Map<String, dynamic> json) {
    return Child(
      id: json['id'] as String,
      parent: json['parent'] as String,
      firstName: json['first_name'] as String,
      lastName: json['last_name'] as String,
      dateOfBirth: DateTime.parse(json['date_of_birth'] as String),
      allowanceAmount: (json['allowance_amount'] as num).toDouble(),
      allowanceFrequency: json['allowance_frequency'] as String,
      points: json['points'] as int,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'parent': parent,
      'first_name': firstName,
      'last_name': lastName,
      'date_of_birth': dateOfBirth.toIso8601String(),
      'allowance_amount': allowanceAmount,
      'allowance_frequency': allowanceFrequency,
      'points': points,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }
}
